lib/ruby_lsp/requests/document_symbol.rb

```diff
@@ -199,6 +216,15 @@ def visit_var_field(node)
 
       private
 
+      sig { params(node: SyntaxTree::Node).returns(T::Boolean) }
+      def test_class?(node)
+        class_node = node.child_nodes[1]
+        return false unless class_node.is_a?(SyntaxTree::ConstPathRef)
+
+        class_name = class_node.constant.value
+        class_name.end_with?("TestCase") || class_name.end_with?("IntegrationTest")
```

Comment:
This should cover all the types provided by Rails:
https://guides.rubyonrails.org/testing.html#a-brief-note-about-test-cases

However if someone defines another kind of test class to inherit from then it could be missed, if named another way. We could possibly make this more generic by matching on anything ending or beginning with `Test`.